### PR TITLE
#1-transports-implementation

### PR DESCRIPTION
**Pull request by agniev-a-hub**

### CODE REVIEW

**Comment by agniev-a-hub** [comment]
getSession, setSession, deleteSession - TBD;

Current progress - WIP on HttpTransport.write() and finish.

File: src/transports/abstract.js [hunk 1]
```diff
@@ -1 +1,30 @@
...
+
+  error(code, { id, name = 'Error', error = null }) {
+    const preparedCode = code || 520;
+    const { server, req, ip } = this;
+    const { console } = server;
+    const { url, method } = req;
+    const status = http.STATUS_CODES[preparedCode];
+    const message = error ? error.message : status || 'Unknown error';
+    const reason = `${status}\t${preparedCode}\t${error ? error.stack : 'Unknown stack'}`;
+    console.error(`IP: ${ip} - Method: ${method} - URL: ${url} - Reason: ${reason}`);
```
Comment (lemarco) [inline src/transports/abstract.js]: should be from logger

File: src/transports/http.js [hunk 2]
```diff
@@ -1,7 +1,61 @@
...
+    const streaming = data instanceof Readable;
+    let mimeType = mime.getType('html');
+    if (httpCode === 200) {
+      const fileType = mime.getType(ext);
+      if (fileType) mimeType = fileType;
+    }
+    const headers = { ...corsOptions, 'Content-Type': mimeType };
+    if (httpCode === 206) {
+      const { start, end, size = '*' } = options;
+      headers['Content-Range'] = `bytes ${start}-${end}/${size}`;
```
Comment (lemarco) [inline src/transports/http.js:35]: i don't think we need this code

File: src/transports/http.js [hunk 3]
```diff
@@ -1,7 +1,61 @@
...
+    res.writeHead(httpCode, headers);
+    res.end(data);
+  }
+
+  getSession() {
+    /* TBD */
+  }
+
+  setSession() {
+    /* TBD */
```
Comment (lemarco) [inline src/transports/http.js]: better to do throw not implemented

File: src/config/config.js [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+  AllowOrigin: ['*'],
+  AllowMethods: ['*'],
+  AllowHeaders: ['*'],
+  ExposeHeaders: ['*'],
+  AllowCredentials: true,
+  MaxAge: 0,
+};
+
+const corsObjectSchema = z.object({
+  AllowOrigin: z.array(z.string()).default(CORS_DEFAULT_CONFIG.AllowOrigin),
```
Comment (lemarco) [inline src/config/config.js]: why are they capitalized?

File: tests/config.spec.js [hunk 5]
```diff
@@ -1,132 +1,141 @@
...
-      const config = {};
-      const result = validateConfig(config);
-      expect(result).toEqual({
-        proto: 'http',
-        transport: ['http'],
-        port: 8888,
-        host: 'localhost',
-        authStrategy: 'none',
-      });
-    });
```
Comment (lemarco) [inline tests/config.spec.js]: ?

File: tests/config.spec.js [hunk 6]
```diff
@@ -1,132 +1,141 @@
...
-      const config = {
-        proto: 'http',
-        transport: ['http', 'ws'],
-        port: 3000,
-        host: '127.0.0.1',
-        authStrategy: 'bearer',
-      };
-      const result = validateConfig(config);
-      expect(result).toEqual(config);
-    });
```
Comment (lemarco) [inline tests/config.spec.js]: ?

**Comment by agniev-a-hub** [comment]
As agreed push without finished tests. 

Will adjust later.

**Pull request merged**